### PR TITLE
Update invalid links in README.md

### PR DESCRIPTION
**Pull request by JeffResc**
## Proposed change

This updates the links to the frontend development page in the README file.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
N/A
```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,12 +6,12 @@ This is the repository for the official [Home Assistant](https://home-assistant.
 
 - [View demo of Home Assistant](https://demo.home-assistant.io/)
 - [More information about Home Assistant](https://home-assistant.io)
-- [Frontend development instructions](https://developers.home-assistant.io/docs/en/frontend_index.html)
+- [Frontend development instructions](https://developers.home-assistant.io/docs/frontend/development.html)
```
Comment (frenck) [inline README.md:9]:
```suggestion
- [Frontend development instructions](https://developers.home-assistant.io/docs/frontend/development/)
```

File: README.md [hunk 2]
```diff
@@ -6,12 +6,12 @@ This is the repository for the official [Home Assistant](https://home-assistant.
...
 
 ## Development
 
 - Initial setup: `script/setup`
-- Development: [Instructions](https://developers.home-assistant.io/docs/en/frontend_development.html)
+- Development: [Instructions](https://developers.home-assistant.io/docs/frontend/development.html)
```
Comment (frenck) [inline README.md:14]:
```suggestion
- Development: [Instructions](https://developers.home-assistant.io/docs/frontend/development/)
```

**Review by frenck — CHANGES_REQUESTED** [review]
The developers' website doesn't use `.html` extension anymore.

**Comment by JeffResc** [comment]
I used those to keep consistency as I saw many links using .html, my apologies, I’ll exclude that from now on

**Review by frenck — APPROVED** [review]
👍  Thanks for updating @JeffResc

**Pull request merged**